src/key_store/mod.rs

```diff
@@ -0,0 +1,184 @@
+//! A storage solution for cryptographic key material.
+//!
+//! This module provides access to the `KeyStore` struct, which manages the
+//! storage of `CredentialBundle` and `KeyPackageBundle` instances. The
+//! development of this module is tracked in #337, which also includes a
+//! roadmap.
+//!
+//! The current key store enables the storage of `CredentialBundle` instances,
+//! and grants access to `CredentialBundle` references via the
+//! `SignaturePublicKey` of the corresponding `Credential`.
+//!
+//! A `KeyStore` is meant to be used across multiple `ManagedGroup` instances to
+//! allow sharing the same `CredentialBundle`. If this is not desired, multiple
+//! `KeyStore` instances can be used across groups.
+//!
```

Comment:
So what I'm missing here is the high level idea of which parts are stored, what's returned as a reference, what's returned as a copy (and why each time).

Reply:
I've extended the documentation to hopefully provide some clarity on this.

Reply:
Can you also add explanations for `generate_key_package` and `generate_credential`?